contracts/CompoundMiddleContract.sol

```diff
@@ -247,65 +182,116 @@ contract CompoundMiddleContract {
         underlying.approve(_cErc20Contract, _numTokensToSupply);
 
```

Comment:
I think we have to use `transferFrom` function as well to transfer the user's ERC20 tokens to address(this).

Also, let's use the safeERC20 library.

Reply:
Oh right, need to transfer to the contract first.